### PR TITLE
Adding precision to loadingbars percentage

### PR DESCRIPTION
**Pull request by Sanduhr32**

### CODE REVIEW

**Comment by Sanduhr32** [comment]
Choosing a precision over 19 will cause issues with the division after casting from `long` to `double`.
Choosing a precision below 0 will always result in 0%

**Comment by Sanduhr32** [comment]
I should have fixed my misstakes while writing while being brain-afk

**Comment by Sanduhr32** [comment]
I suggest squash merging

File: src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java [hunk 1]
```diff
@@ -78,6 +83,18 @@ public LoadingBarConfig setBorderWidth(int borderWidth) {
...
+    
+    public int getPrecision() {
+        return precision;
+    }
+    
+    public LoadingBarConfig setPrecision(int precision) {
+        if (precision >= 2 && precision < 20)
+            this.precision = precision;
+        else
+            this.precision = 19;
```
Comment (duncte123) [inline src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java]: brackets around the if else

File: src/main/java/me/duncte123/loadingbar/LoadingBar.java [hunk 2]
```diff
@@ -35,6 +35,11 @@ private LoadingBar() {
      * @return The percentage of how much has passed from this year
      */
     public static double getPercentage() {
+        return getPercentage(19);
+    }
+    
+    public static double getPercentage(int precision) {
+        if (precision > 19 || precision < 2) precision = 19;
```
Comment (duncte123) [inline src/main/java/me/duncte123/loadingbar/LoadingBar.java]:
```suggestion
        if (precision > 19 || precision < 2) {
            precision = 19;
        }
```

File: src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java [hunk 3]
```diff
@@ -43,10 +44,14 @@
      * @param fillColor
      *         The color of the fill (green part)
      */
-    public LoadingBarConfig(int width, int height, int borderWidth, Color borderColor, Color innerColor, Color fillColor) {
+    public LoadingBarConfig(int width, int height, int borderWidth, int precision, Color borderColor, Color innerColor, Color fillColor) {
         this.width = width;
         this.height = height;
         this.borderWidth = borderWidth;
+        if (precision >= 2 && precision < 20)
```
Comment (duncte123) [inline src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java]: missing brackets around the if statement

File: src/test/java/me/duncte123/loadingbartestclass/Main.java [hunk 4]
```diff
@@ -40,7 +40,8 @@ public static void main(String[] args) {
 
         LoadingBarConfig config = LoadingBarConfig.defaultConfig()
                 .setFillColor(Color.RED)
-                .setBorderColor(Color.PINK);
+                .setBorderColor(Color.PINK)
+                .setPrecision(3);
```
Comment (duncte123) [inline src/test/java/me/duncte123/loadingbartestclass/Main.java:44]: Please re-render the example with this config

**Review by duncte123 — CHANGES_REQUESTED** [review]

File: src/main/java/me/duncte123/loadingbar/LoadingBar.java [hunk 5]
```diff
@@ -35,6 +35,13 @@ private LoadingBar() {
      * @return The percentage of how much has passed from this year
      */
     public static double getPercentage() {
+        return getPercentage(19);
+    }
+    
+    public static double getPercentage(int precision) {
+        if (precision > 19 || precision < 2) {
+            precision = 19;
+        }
```
Comment (duncte123) [inline src/main/java/me/duncte123/loadingbar/LoadingBar.java:45]: Add an enter here

File: src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java [hunk 6]
```diff
@@ -43,10 +44,15 @@
      * @param fillColor
      *         The color of the fill (green part)
      */
-    public LoadingBarConfig(int width, int height, int borderWidth, Color borderColor, Color innerColor, Color fillColor) {
+    public LoadingBarConfig(int width, int height, int borderWidth, int precision, Color borderColor, Color innerColor, Color fillColor) {
         this.width = width;
         this.height = height;
         this.borderWidth = borderWidth;
+        if (precision >= 2 && precision < 20) {
```
Comment (duncte123) [inline src/main/java/me/duncte123/loadingbar/LoadingBarConfig.java]: Add an enter above and below if-statement

**Review by duncte123 — CHANGES_REQUESTED** [review]

File: src/test/java/me/duncte123/loadingbartestclass/Main.java (hunk 4 above)
Comment (Sanduhr32) [reply src/test/java/me/duncte123/loadingbartestclass/Main.java:44]: The new rendered file is not detected as a changed file

**Review by duncte123 — APPROVED** [review]

**Pull request merged**